cms/djangoapps/contentstore/views/tests/test_item.py

```diff
@@ -2469,6 +2470,42 @@ def verify_dndv2_present(support_level):
         self.assertIsNone(get_xblock_problem('Staff Graded Points'))
         self.assertIsNone(get_xblock_problem('Drag and Drop'))
 
+
+    def test_discussion_button_present_no_provider(self):
+        """
+        Test the Discussion button present when no discussion provider configured for course
+        """
+        templates = get_component_templates(self.course)
+        button_names = [template['display_name'] for template in templates]
+        self.assertIn('Discussion', button_names)
+
+    def test_discussion_button_present_legacy_provider(self):
+        """
+        Test the Discussion button present when legacy discussion provider configured for course
+        """
+        course_key = self.course.location.course_key
+
+        #Create a discussion configuration with discussion provider set as legacy
```

Comment:
nit: Please add one space between the `#` and the comment. 
```suggestion
        # Create a discussion configuration with discussion provider set as legacy
```